packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/feature_counter/views/counter_page.dart

```diff
@@ -24,45 +24,66 @@ class CounterPage extends StatelessWidget implements AutoRouteWrapper {
 
   @override
   Widget wrappedRoute(BuildContext context) => MultiProvider(
-        providers: CounterDependencies.of(context).providers,
-        child: this,
-      );
+    providers: CounterDependencies.of(context).providers,
+    child: this,
+  );
 
   @override
   Widget build(BuildContext context) => Scaffold(
-        appBar: AppBar(title: Text(context.l10n.counterPageTitle)),
-        body: Center(
-          child: Column(
-            mainAxisAlignment: MainAxisAlignment.center,
-            children: <Widget>[
-              _buildErrorListener(),
-              _buildCount(),
-            ],
+    appBar: _buildAppBar(context),
+    body: Center(
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        children: [
+          _buildErrorListener(),
+          RxBlocBuilder<CounterBlocType, int>(
+            state: (bloc) => bloc.states.count,
+            builder: (context, countState, bloc) =>
+                _buildCount(context, countState),
           ),
+        ],
+      ),
+    ),
+    floatingActionButton: _buildActionButtons(context),
+  );
+
+  AppBar _buildAppBar(BuildContext context) => AppBar(
+    title: Text(context.l10n.counterPageTitle),
+    actions: [
+      RxBlocBuilder<CounterBlocType, bool>(
+        state: (bloc) => bloc.states.isLoading,
+        builder: (context, loadingState, bloc) => IconButton(
```

Comment:
Two questions:
1. Is there any reason we don't use the design system?
2. Is there any reason this button is not built as part of the app's widget catalogue?